src/GeositeFramework/js/Export.js

```diff
@@ -54,63 +68,79 @@ require(['use!Geosite'],
-                                // "stick", regardless of what's in the stylesheet
-                                $("#legend-container-0").css({ visibility: "hidden" });
-                            }
-                            var pageOrientation = $("[name='export-orientation']").filter(function () { return this.checked; }).map(function () { return this.value; }).first()[0];
-                            var orientDeferred = $.Deferred();
 
+                            var pageOrientation = _.head($("[name='export-orientation']")
+                                .filter(function () { return this.checked; })
+                                .map(function () { return this.value; })
+                                .first());
                             if (pageOrientation === "Landscape") {
                                 $('<link>', {
                                     rel: 'stylesheet',
                                     href: 'css/print-landscape.css',
                                     'class': '.print-orientation-css',
                                 }).appendTo('head');
-                                _.delay(orientDeferred.resolve, 750);
-                                $(".legend-layer").each(function(el) { $(this).children(".item").wrapAll("<div class='print-legend-coll'></div>"); });
+                                _.delay(orientDeferred.resolve, 500);
                             } else {
                                 $('<link>', {
                                     rel: 'stylesheet',
                                     href: 'css/print-portrait.css',
                                     'class': '.print-orientation-css',
                                 }).appendTo('head');
-                                _.delay(orientDeferred.resolve, 750);
+                                _.delay(orientDeferred.resolve, 500);
                             }
 
                             $.when(orientDeferred).then(function () {
-                                window.print();
-                                previewDeferred.resolve();
+                                map.width = parseFloat($("#export-print-preview-map").css("width"));
+                                map.height = parseFloat($("#export-print-preview-map").css("height"));
+                                map.resize(true);
+                                map.centerAt(mapDimensions.extent.getCenter());
+                                _.delay(resizeDeferred.resolve, 500);
+
+                                // the legend items are affected by the map resize, so
+                                // they are manipulated just before printing
+                                $.when(resizeDeferred).then(function() {
+                                    // expand legend container if minimized
+                                    if ($("#legend-container-0").hasClass("minimized")) {
+                                        _.head($(".legend-close")).click();
```

Comment:
The use of `_.head` here and below is superfolous, jQuery operates on collections natively.